### PR TITLE
george / rm78462 / add signup wizard component

### PR DESCRIPTION
**Pull request by george-usynin-binary**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   add wizard component into cashier

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { Text } from '@deriv/components';
+import { Localize, localize } from '@deriv/translations';
+import { Wizard } from '@deriv/ui';
+
+type TSignupWizardProps = {
+    setIsWizardOpen: (is_wizard_open: boolean) => void;
```
Comment (farzin-deriv) [inline packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:7]: @george-usynin-binary It's not a good idea to make this component aware of outside, I suggest having callbacks like `onOpen` or `onClose` and letting the parent component handle the logic.

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+const SignupWizard = ({ setIsWizardOpen }: TSignupWizardProps) => {
+    const [create_app_state, setCreateAppState] = React.useState({});
```
Comment (farzin-deriv) [inline packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:11]:
```suggestion
```
@george-usynin-binary It's unused 🤔

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    };
+
+    return (
+        <Wizard
+            has_dark_background
+            lock_final_step={false}
+            onClose={closeWizardHandler}
+            onComplete={handleComplete}
+            wizard_title={localize('Become a payment agent')}
+            primary_button_label={is_final_step ? localize('Submit') : localize('Next')}
```
Comment (farzin-deriv) [inline packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:35]:
```suggestion
            primary_button_label={current_step === 3 ? localize('Submit') : localize('Next')}
```
@george-usynin-binary Couldn’t we just simply check for the last index? 🤔

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx (hunk 2 above)
Comment (george-usynin-binary) [reply packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:11]: @farzin-deriv for now yes, i have made a mocked wizard, but i better remove it, to not confuse

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx (hunk 3 above)
Comment (george-usynin-binary) [reply packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:35]: @farzin-deriv some steps can be hidden and the last index will not be the last step, using step_key is mentioned in library(documentation/usage) itself, that's why i use step_key

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx (hunk 1 above)
Comment (george-usynin-binary) [reply packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:7]: @farzin-deriv agree with you, will refactor it

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    const onChangeStep = (_current_step: number, _current_step_key?: string) => {
+        setCurrentStepKey(_current_step_key);
+    };
+
+    return (
+        <Wizard
+            has_dark_background
+            lock_final_step={false}
+            onClose={closeWizardHandler}
+            onComplete={handleComplete}
```
Comment (farzin-deriv) [inline packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:31]:
```suggestion
            onClose={onClose}
            onComplete={onClose}
```

**Review by farzin-deriv — APPROVED** [review]

File: packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx (hunk 4 above)
Comment (farzin-deriv) [reply packages/cashier/src/pages/payment-agent/payment-agent-signup/signup-wizard/signup-wizard.tsx:31]: @george-usynin-binary Just ignore it as we discussed 🙏🏻

**Review by vinu-deriv — APPROVED** [review]

**Pull request merged**